packages/shared/src/modules/users/libs/enums/user-validation-regex-rule.enum.ts

```diff
@@ -1,9 +1,10 @@
 import { UserValidationRule } from "./user-validation-rule.enum.js";
 
 const UserValidationRegexRule = {
-	EMAIL_DOMAIN_MIN_MAX_CHARS: /@.{4,35}$/,
-	EMAIL_DOMAIN_PART_VALID_CHARS: /(?<=@)[\dA-Za-z-]{4,35}(\.[\dA-Za-z-]{2,})+$/,
-	EMAIL_LOCAL_PART_VALID_CHARS: /^[\w!#$&'*+=\\-]{1,35}(?=@)/,
+	EMAIL_DOMAIN_PART_VALID_CHARS:
+		/^[^\s@]+@(?=.{4,35}$)[\dA-Za-z]+(?:-[\dA-Za-z]+)*\.[\dA-Za-z]{2,}$/,
+	EMAIL_LOCAL_PART_VALID_CHARS:
+		/^(?!\.)((?!\.{2})[\w!#$&'*+.=\\-]){1,35}(?<!\.)@/,
```

Comment:
Why do we need all of that? Why not just use `z.string().email()`?

Reply:
Mention from the requirements document we accept several special character such as ( * # $ ! = & '_+ –)   in local part of email. We can't include special character if we using `.email()`. For example I try to validation this email below using `.email()`
![image](https://github.com/user-attachments/assets/75769f6f-aea6-42b1-9ab4-9d340bf42779)